tensorflow_probability/python/mcmc/dual_averaging_step_size_adaptation.py

```diff
@@ -0,0 +1,299 @@
+"""DualAveragingStepSizeAdaptation TransitionKernel."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+
+import numpy as np
+import tensorflow as tf
+
+from tensorflow_probability.python.internal import dtype_util
+from tensorflow_probability.python.mcmc import kernel as kernel_base
+from tensorflow_probability.python.mcmc.internal import util as mcmc_util
+
+
+@mcmc_util.make_innermost_setter
+def _hmc_like_step_size_setter_fn(kernel_results, new_step_size):
+  return kernel_results._replace(
+    accepted_results=kernel_results.accepted_results._replace(
+      step_size=new_step_size
+    )
```

Comment:
nit: Can you place these closing parens on the line above. In TFP we dont follow the open-paren style you use here. (Although it does have its merits, to be sure!)